pkg/ccl/changefeedccl/sink_kafka.go

```diff
@@ -257,10 +257,21 @@ func (s *kafkaSink) Dial() error {
 		return err
 	}
 
-	if err = client.RefreshMetadata(s.Topics()...); err != nil {
-		// Now that we do not fetch metadata for all topics by default, we try
-		// RefreshMetadata manually to check for any connection error.
-		return errors.CombineErrors(err, client.Close())
+	if err = client.RefreshMetadata(); err != nil {
+		// Dial() seems to be a weird state while topics are still setting up and
+		// calling RefreshMetadata with specific topics s.Topics() here triggered a
+		// sarama error (pq: kafka server: Replication-factor is invalid). To match
+		// the previous behaviour in sarama prior to #114740 during Dial(), we
+		// manually fetch metadata for all topics just during Dial(). See more in
+		// #116872.
+		if errors.Is(err, sarama.ErrLeaderNotAvailable) || errors.Is(err, sarama.ErrReplicaNotAvailable) ||
```

Comment:
Alternatively, we could temporarily change config.Metadata.Full to true temporarily during Dial() and change it back to config.Metadata.Full to false after Dial() if we think this way is too hacky.

Reply:
I'm okay with the current implementation. To be safe, I would prefer to always have `config.Metadata.Full=false`.